src/PreCICE.jl

```diff
@@ -1332,5 +1340,159 @@ function actionReadIterationCheckpoint()
     return unsafe_string(msgCstring)
 end
 
+@doc """
+
+    isGradientDataRequired(dataID::Integer)::Bool
+        
+Checks if the given data set requires gradient data. We check if the data object has been intialized with the gradient flag.
+# Arguments
+- `dataID::Integer`: ID of the data to be checked. Obtained by [`getDataID`](@ref).
+
+"""
+function isGradientDataRequired(dataID::Integer)
+        return ccall((:precicec_isGradientDataRequired, "libprecice"), Cint, (Cint,), dataID)
+end
+
+@doc """
+
+    writeBlockVectorGradientData(dataID::Integer, valueIndices::AbstractArray{Cint}, gradientValues::AbstractArray{Float64})
+
+
+Write gradient data of a vector data as a block, the value of a specified vertices to a dataID.
+
+# Arguments
+- `dataID::Integer`: ID of the data to be written. Obtained by [`getDataID`](@ref).
+- `valueIndices::AbstractArray{Cint}`: Indices of the vertex.
+- `gradientValues::AbstractArray{Float64}`: The gradient values to write.
+
+# Notes
+
+Previous calls:
+ - [`initialize`](@ref) 
+
+# Examples
+
+Write gradient values of a vector data for a 2D problem with 2 vertices:
+
+```julia
+data_id = 1
+valueIndices = [1,2,3]
+gradientValues = [1.0, 2.0, 3.0, 4.0, 5.0, 6.0, 7.0, 8.0]
```

Comment:
Shouldn't this be a 3D tensor with dimensions [N, Dimension, Dimension] or something similar?

Reply:
No, it is always 2D matrices [NxD] where D yields dimension squared.

```julia
valueIndices = [1,2]
gradientValues = [1.0 2.0 3.0 4.0; 5.0 6.0 7.0 8.0]
```

This would be the correct form